app/services/resync_service.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+
+class ResyncService < ApplicationService
+  attr_reader :document
+
+  def initialize(document)
+    @document = document
+  end
+
+  def call
+    update_live_edition(document.live_edition) if document.live_edition
+    update_current_edition(document.current_edition) unless
+      document.current_edition == document.live_edition
+  end
+
+private
+
+  def update_live_edition(edition)
+    edition.update!(
+      revision_synced: false,
+      system_political: PoliticalEditionIdentifier.new(edition).political?,
+      government_id: government_id(edition, document),
+    )
+
+    PreviewService.call(
+      edition,
+      update_type: "republish",
+      bulk_publishing: true,
+    )
+
+    PublishAssetService.call(edition, nil)
+
+    if edition.withdrawn?
+      withdraw(edition)
+    elsif edition.removed?
+      redirect_or_remove(edition)
+    else
+      publish(edition)
+    end
+
+    edition.update!(revision_synced: true)
+  end
+
+  def update_current_edition(edition)
+    edition.update!(
+      revision_synced: false,
+      system_political: PoliticalEditionIdentifier.new(edition).political?,
```

Comment:
We should set government on this too - in case the content was already published or is backdated.